run_all_experiments.py

```diff
@@ -351,6 +358,32 @@ def add_to_json_report(outdir: str, key: str, value: Any) -> None:
     f.write(json.dumps(json_report))
 
 
+def _process_total_coverage_gain(results: list[Result]) -> dict[str, float]:
+  """Process and calculate the total coverage gain for each project."""
+  textcov_dict: dict[str, list[textcov.Textcov]] = {}
+  for result in results:
+    if not isinstance(result.result, run_one_experiment.AggregatedResult):
+      continue
```

Comment:
Did you mean `result.result is None`?
A more specific check is preferred if possible : )

Reply:
No, this is added as a remedy to past the type checking because the result.result is could be a str as suggested by the type hints on line 64.